src/people.rs

```diff
@@ -167,6 +167,68 @@ seq!(Z in 1..20 {
     impl_query!(Z);
 });
 
+pub trait Tabulator {
+    fn setup(&self, context: &mut Context);
+    fn get_typelist(&self) -> Vec<TypeId>;
+    fn get_columns(&self) -> Vec<String>;
+}
+
+impl<T: PersonProperty + 'static> Tabulator for (T,) {
+    fn setup(&self, context: &mut Context) {
+        context.index_property_noargs::<T>();
+    }
+    fn get_typelist(&self) -> Vec<TypeId> {
+        vec![std::any::TypeId::of::<T>()]
+    }
+    fn get_columns(&self) -> Vec<String> {
+        vec![String::from(
+            std::any::type_name::<T>().split("::").last().unwrap(),
```

Comment:
You actually can't guarantee that types will be named like this, as it's a compiler internal. `type_name::<T>()` could be legally implemented as `format!("Banana|{:?}", TypeId::of::<T>())`. I think instead we need to extend `PersonProperty` to have a name field.